### ISSUE TITLE
cheetah-flake pre-commit hook fails with AttributeError: 'FlakesChecker' object has no attribute 'JOINEDSTR' when run on .tmpl file containing py3.6 f-string

### ISSUE THREAD

**Issue by dan98765** [asottile/cheetah_lint#23]
From running pre-commit on a .tmpl file with py3.6 f-string (f'abcd {somevariable} xyz') with .pre-commit-config.yaml cheetah_lint sha v0.3.4. 

Traceback:
```
Traceback (most recent call last):
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/bin/cheetah-flake", line 11, in <module>
    load_entry_point('cheetah-lint==0.3.4', 'console_scripts', 'cheetah-flake')()
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/cheetah_lint/flake.py", line 408, in main
    retv |= flake(filename)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/cheetah_lint/flake.py", line 394, in flake
    flakes = get_flakes(file_contents)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/cheetah_lint/flake.py", line 387, in get_flakes
    data += get_from_py(file_contents)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/cheetah_lint/flake.py", line 306, in get_from_py
    check(py_lines), py_lines, cheetah_lines,
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/cheetah_lint/flake.py", line 261, in check_flake8
    checker.input_file('<compiled cheetah>', py_lines)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/flake8/engine.py", line 195, in input_file
    line_offset=line_offset,
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/flake8/engine.py", line 172, in _retry_serial
    return func(*args, **kwargs)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/flake8/engine.py", line 126, in input_file
    return fchecker.check_all(expected=expected, line_offset=line_offset)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pycodestyle.py", line 1608, in check_all
    self.check_ast()
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pycodestyle.py", line 1554, in check_ast
    checker = cls(tree, self.filename)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/flake8/_pyflakes.py", line 63, in __init__
    withDoctest=withDoctest)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 426, in __init__
    self.runDeferred(self._deferredFunctions)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 463, in runDeferred
    handler()
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 1055, in runFunction
    self.handleNode(stmt, node)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 794, in handleNode
    handler(node)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 747, in handleChildren
    self.handleNode(node, tree)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 793, in handleNode
    handler = self.getNodeHandler(node.__class__)
  File "/nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/checker.py", line 627, in getNodeHandler
    self._nodeHandlers[node_class] = handler = getattr(self, nodeType)
AttributeError: 'FlakesChecker' object has no attribute 'JOINEDSTR'
```

I was hesitant to post this here since it's not actually code in cheetah-flake that's the problem, exactly.

The error is being thrown by pyflakes. Googling the error, turns out this was an error in older versions of pyflakes and is fixed in newer versions. I checked which version of pyflakes is the one running:
```
(virtualenv_run) $ cat /nail/home/daniellg/.cache/pre-commit/repo55sxfmr_/py_env-python3.6/lib/python3.6/site-packages/pyflakes/__init__.py 
__version__ = '1.2.3'
```
Which is old enough it still has lack of py36 support. 

cheetah-lint is calling flake8, which itself is making this call to the old version of pyflakes. I quickly looked at repos for pyflakes/flake8 and didn't see old versions pinned, nor are they pinned in this cheetah_lint repo. So I'm not sure yet why it's using v1.2.3 of pyflakes.

As a sanity check, I did nuke /nail/home/daniellg/.cache/pre-commit/ completely and reinstall hooks but no change in behavior.

**Comment by asottile** [comment]
Unfortunately there's not really much that can be done here due to https://github.com/asottile/cheetah_lint/issues/14 :'(

**Comment by dan98765** [comment]
@asottile thanks for the quick response, despite the sad news.

**Comment by asottile** [comment]
@dan98765 if you want to try and fix it, the suggestion in #14 will probably work (and won't be too difficult) -- the code hint is here: https://github.com/asottile/cheetah_lint/blob/d4e9471b3647d1ccfb8e5070dc7358b7f95f4a36/cheetah_lint/flake.py#L257-L264
